lib/solidus_bolt/engine.rb

```diff
@@ -13,13 +13,34 @@ class Engine < Rails::Engine
 
     initializer "solidus_bolt.add_static_preference", after: "spree.register.payment_methods" do |app|
       app.config.spree.payment_methods << SolidusBolt::PaymentMethod
-      Spree::Config.static_model_preferences.add(
-        SolidusBolt::PaymentMethod,
-        'bolt_credentials', {
-          bolt_api_key: ENV['BOLT_API_KEY'],
-          bolt_signing_secret: ENV['BOLT_SIGNING_SECRET']
-        }
-      )
+      begin
+        bolt_config = SolidusBolt::BoltConfiguration.fetch
+      rescue ActiveRecord::StatementInvalid
+        bolt_config = nil
+      ensure
+        Spree::Config.static_model_preferences.add(
+          SolidusBolt::PaymentMethod,
+          'bolt_credentials', {
+            bolt_api_key: ENV['BOLT_API_KEY'],
+            bolt_signing_secret: ENV['BOLT_SIGNING_SECRET'],
+            bolt_publishable_key: ENV['BOLT_PUBLISHABLE_KEY'],
+            bolt_environment: ENV['BOLT_ENVIRONMENT'],
+          }
+        )
+        Spree::Config.static_model_preferences.add(
+          SolidusBolt::PaymentMethod,
+          'bolt_config_credentials', {
+            bolt_api_key: bolt_config&.api_key,
+            bolt_signing_secret: bolt_config&.signing_secret,
+            bolt_publishable_key: bolt_config&.publishable_key,
+            bolt_environment: bolt_config&.environment,
+          }
+        )
```

Comment:
the block feels a bit long to me - wondering if extracting each static_model_preference into a private method would improve readability

Reply:
We can probably do that

Reply:
moved these into private methods now

Reply:
looks more readable now